src/main/java/org/simplify4u/plugins/utils/PublicKeyUtils.java

```diff
@@ -118,7 +118,7 @@ public static String keyIdDescription(PGPPublicKey publicKey, PGPPublicKeyRing p
             return Optional.empty();
         }
 
-        Iterator signatures = publicKey.getSignaturesOfType(PGPSignature.SUBKEY_BINDING);
+        Iterator<?> signatures = publicKey.getSignaturesOfType(PGPSignature.SUBKEY_BINDING);
```

Comment:
sonar rules?

Reply:
Sure we can do this in Sonar rules. My reason for choosing to expose them as compiler warnings, is that doing these wrong, e.g. not adding generic types (even if just `<?>`) means you're using Java 1.4 or earlier coding style. I see it as setting the bar at the right level and avoiding building if minimum level is not reached. You get earlier feedback, as sonar does not run on every build.

So, let me know what you prefer. In the end, either way gives the feedback.

Reply:
we can have both - compiler warning and sonar rules

Reply:
I checked, is already enabled in Sonar: java:S3740.